virt/virtual_machines/virt-accessing-vm-ssh.adoc

```diff
@@ -0,0 +1,120 @@
+
+Static key::
+* Added to the VM at startup.
+* Uses the cloud-init configuration drive and does not affect cloud-init user data.
+* Can be added when you create a VM by using the xref:../../virt/virtual_machines/virt-accessing-vm-ssh.adoc#adding-static-public-key-vm-web_virt-accessing-vm-ssh[web console] or the xref:../../virt/virtual_machines/virt-accessing-vm-ssh.adoc#adding-static-public-key-project-web_virt-accessing-vm-ssh[command line].
+* Can be added to a xref:../../virt/virtual_machines/virt-accessing-vm-ssh.adoc#adding-static-public-key-project-web_virt-accessing-vm-ssh[project] by using the web console. Afterwards, the key is automatically added to the VMs that you create in this project.
+
+Dynamic key::
+* Can be added or revoked while the VM is running.
+* Can be disabled for security reasons. If the setting is disabled, the VM inherits the key setting of the image from which it was created.
+* Uses the QEMU guest agent.
+* Only supported by {op-system-base-full} 9.
+
+include::modules/virt-adding-static-public-key-vm-web.adoc[leveloffset=+3]
+
+include::modules/virt-adding-static-public-key-cli.adoc[leveloffset=+3]
+
+include::modules/virt-adding-static-public-key-project-web.adoc[leveloffset=+3]
+
+// dynamic ssh key configuration will go here
+
+include::modules/virt-running-virtctl-ssh-command.adoc[leveloffset=+2]
+
+.Copying the `virtctl ssh` command
+
+You can copy the `virtctl ssh` command in the web console by selecting *Copy SSH command* from the options {kebab} menu beside a VM on the xref:../../virt/getting_started/virt-web-console-overview.adoc#virtualmachines-page_virt-web-console-overview[*VirtualMachines* page].
+
+include::modules/virt-using-openssh-and-virtctl-port-forward.adoc[leveloffset=+1]
+
+[id="using-services-ssh_virt-accessing-vm-ssh"]
+== Using a service for SSH access
+
+You can create a service for a virtual machine (VM) and connect to the IP address and port exposed by the service.
+
+Services provide excellent performance and are recommended for applications that are accessed from outside the cluster or within the cluster. Ingress traffic is protected by firewalls.
+
+If the cluster network cannot handle the traffic load, consider using a secondary network for VM access.
+
+include::modules/virt-about-services.adoc[leveloffset=+2]
+
+[id="creating-services-ssh_virt-accessing-vm-ssh"]
+=== Creating a service
+
+You can create a service to expose a virtual machine (VM) by using the {product-title} web console, `virtctl` command line tool, or a YAML file.
+
+include::modules/virt-creating-service-web.adoc[leveloffset=+3]
+
+include::modules/virt-creating-service-virtctl.adoc[leveloffset=+3]
+
+After you create a service with `virtctl`, you must add `special: key` to the `spec.template.metadata.labels` stanza of the `VirtualMachine` manifest. See xref:../../virt/virtual_machines/virt-accessing-vm-ssh.adoc#virt-creating-service-cli_virt-accessing-vm-ssh[Creating a service by using the command line].
```

Comment:
It might help make the follow up action more visible by creating a [Next steps](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies) title.